state_machine/applications/system-check/tests/nvm_access_test.py

```diff
@@ -0,0 +1,155 @@
+from lib.pycubed import _Satellite, cubesat
+from lib.bitflags import bitFlag, multiBitFlag, multiByte
+from print_utils import bold, normal
+
+
+def get_nvm_flags():
+    """
+    Find all of the bitFlags in the
+    _Satellite class.
+    """
+
+    sat_vars = vars(_Satellite)
+    nvm_flags = []
+    for v in sat_vars:
+        if type(sat_vars[v]) == bitFlag:
+            nvm_flags.append(v)
+
+    return nvm_flags
+
+def get_nvm_counters():
+    """
+    Find all of the multiBitFlag and multiByte counters in the
+    _Satellite class.
+    """
+
+    sat_vars = vars(_Satellite)
+    nvm_counters = []
+    for v in sat_vars:
+        if type(sat_vars[v]) == multiBitFlag or type(sat_vars[v]) == multiByte:
+            nvm_counters.append(v)
+
+    return nvm_counters
+
+def verify_value(field_str, value):
+    setattr(cubesat, field_str, value)
+    if not getattr(cubesat, field_str) == value:
+        return False
+    return True
+
+def verify_counter(counter_str):
+    """ verify that the counter values can be changed and stay in range """
+
+    # save current counter value
+    counter_prev = getattr(cubesat, counter_str)
+    maxval = vars(_Satellite)[counter_str].maxval
+
+    success = True
+    success &= verify_value(counter_str, 0)
```

Comment:
Can start with ` success = verify_value(counter_str, 0)` instead of `success=True` and `    success &= verify_value(counter_str, 0)`